comment_iowriter.go

```diff
@@ -56,3 +59,86 @@ func (mc *UnifiedCommentWriter) Post(_ context.Context, c *Comment) error {
 	_, err := fmt.Fprintln(mc.w, s)
 	return err
 }
+
+var _ CommentService = &RDJSONLCommentWriter{}
+
+// RDJSONLCommentWriter
+type RDJSONLCommentWriter struct {
+	w io.Writer
+}
+
+func NewRDJSONLCommentWriter(w io.Writer) *RDJSONLCommentWriter {
+	return &RDJSONLCommentWriter{w: w}
+}
+
+func (cw *RDJSONLCommentWriter) Post(_ context.Context, c *Comment) error {
+	if c.ToolName != "" && c.Result.Diagnostic.GetSource().GetName() == "" {
+		c.Result.Diagnostic.Source = &rdf.Source{
+			Name: c.ToolName,
+		}
```

Comment:
```suggestion
		c.Result.Diagnostic.Source.Name = c.ToolName
```

https://github.com/reviewdog/reviewdog/blob/7b581a6cff6cbcc171ab58488b5ad152a26f084d/comment_iowriter.go#L75
It seems to assume that `c.Result.Diagnostic.Source` is not `nil`.
Therefore, I think it would be better to assign `Name` instead of assigning `Source`.

Reply:
It should be ok as-is. `GetXXX` works even if a receiver is nil and name is expected to be non-empty if Source is non-empty. 
https://github.com/reviewdog/reviewdog/blob/7b581a6cff6cbcc171ab58488b5ad152a26f084d/proto/rdf/reviewdog.pb.go#L669-L674